File: src/easy/fizz-buzz/fizz-buzz.ts
export function fizzBuzz(n: number): string[] {
    let result: string[] = [];
    if (n <= 0) {
        return result;
    }
    let isDivBy3 = n % 3 === 0;
    let isDivBy5 = n % 5 === 0;
    if (isDivBy3 && isDivBy5) {
        result.push("FizzBuzz");
    } else if (isDivBy3) {
        result.push("Fizz");
    } else if (isDivBy5) {
        result.push("Buzz");
    } else {
        result.push(String(n));
    }
    return fizzBuzz(n - 1).concat(result);
}

File: src/easy/fizz-buzz/FizzBuzz.swift
class Solution {
    func fizzBuzz(_ n: Int) -> [String] {
        var result = [String]()
        for i in 1...n {
            switch (i % 3, i % 5) {
            case (0, 0):
                result.append("FizzBuzz")
            case (0, _):
                result.append("Fizz")
            case (_, 0):
                result.append("Buzz")
            default:
                result.append(String(i))
            }
        }
        return result
    }
}
